tests/tests/test-xcm/test-xcmv3-max-weight-instructions.ts

```diff
@@ -0,0 +1,222 @@
+import "@moonbeam-network/api-augment";
+import { expect } from "chai";
+import { CHARLETH_ADDRESS } from "../../util/accounts";
+import { describeDevMoonbeam } from "../../util/setup-dev-tests";
+import {
+  XcmFragment,
+  injectHrmpMessage,
+  RawXcmMessage,
+  sovereignAccountOfSibling,
+} from "../../util/xcm";
+import { expectEVMResult } from "../../util/eth-transactions";
+import { ALITH_TRANSACTION_TEMPLATE, createTransaction } from "../../util/transactions";
+
+describeDevMoonbeam(
+  "XCM V3 - Max Weight Instructions",
+  (context) => {
+    let dotAsset: any;
+    let amount: bigint;
+    let paraId: number;
+
+    before("Set up initial constants", async function () {
+      paraId = 888;
+      const paraSovereign = sovereignAccountOfSibling(context, paraId);
+      const metadata = await context.polkadotApi.rpc.state.getMetadata();
+      const balancesPalletIndex = (metadata.asLatest.toHuman().pallets as Array<any>).find(
+        (pallet) => pallet.name === "Balances"
+      ).index;
+
+      // Send some native tokens to the sovereign account of paraId (to pay fees)
+      const { result } = await context.createBlock(
+        createTransaction(context, {
+          ...ALITH_TRANSACTION_TEMPLATE,
+          value: 1_000_000_000_000_000_000,
+          to: paraSovereign,
+          data: "0x",
+        })
+      );
+      expectEVMResult(result.events, "Succeed");
+
+      amount = 1_000_000_000_000_000n;
+      dotAsset = {
+        assets: [
+          {
+            multilocation: {
+              parents: 0,
+              interior: {
+                X1: { PalletInstance: balancesPalletIndex },
+              },
+            },
+            fungible: amount,
+          },
+        ],
+        // weight_limit: new BN(4000000000),
+        beneficiary: CHARLETH_ADDRESS,
+      };
+    });
+
+    // WEIGHT::MAX instructions
+    it("Should not execute UniversalOrigin", async function () {
+      const xcmMessage = new XcmFragment(dotAsset)
+        .withdraw_asset()
+        .buy_execution()
+        .universal_origin({ Parachain: paraId })
+        .as_v3();
+
+      // Mock the reception of the xcm message
+      await injectHrmpMessage(context, paraId, {
+        type: "XcmVersionedXcm",
+        payload: xcmMessage,
+      } as RawXcmMessage);
+      await context.createBlock();
+
+      // Search for WeightNotComputable error
+      const records = (await context.polkadotApi.query.system.events()) as any;
```

Comment:
this is destroying the `FrameSystemEventRecord` type, instead you should preserve it.

You can use the `api.events.<pallet>.<method>.is(<event>)` function to assert whether an event is something or not. e.g.

```
 const events = (await context.polkadotApi.query.system.events()).filter(({event}) =>
            context.polkadotApi.events.xcmpQueue.Fail.is(event)
          );
```